chore: add information about the new signOut redirect url option (#7938)
aws-amplify/docs: src/pages/gen1/[platform]/build-a-backend/auth/add-social-provider/index.mdx

Reviewer: Sanity check: Gen2 doc doesn't have an equivalent page for adding this information?

Author: I couldn't find any relevant place. Only relatable location is the signOut [page](https://docs.amplify.aws/react-native/build-a-backend/auth/connect-your-frontend/sign-out/) but that's for the common use-case and Gen2 has nothing on signInWithRedirect I believe.

Author: Added it [here](https://github.com/aws-amplify/docs/pull/7938/files#diff-d59fe95d17fcc94afb4be025157ed7e88a6b2fb97f15755da89c6127e1b19c53) for Gen2.

</Block>
</BlockSwitcher>

> ### Redirect URLs

Reviewer 2: I think this section is redundant. By examining lines `744` and `746`, I interpret that the change we are currently implementing is the inclusion of multiple redirects, which was actually a feature offered by v6 upon its release.

Author: On RN page, we don't have any info on redirects so I thought this will set up some context before diving into multiple redirects.

> You can provide multiple _Sign in Redirect URI(s)_ & _Sign out redirect URI(s)_.

Reviewer 3: ```suggestion
You can provide multiple _Sign in_ & _Sign out_ redirect URI(s).
```

If you have multiple redirect URI inputs, you'll need to pass them in your Amplify configuration. For example:

> ```javascript

Author: Yes

Amplify.configure({
  Auth: {
    Cognito: {
      loginWith: {
        oauth: {
          redirectSignIn: [
            'myDevApp://'
          ],
          redirectSignOut: [
            'myDevApp://',
            'myProdApp://'
          ],
          ...oauthConfig
        }
      },
      ...userPoolConfig
    }
  }
});
```

> #### Option to specify a redirect URL upon signOut

Reviewer 3: ```suggestion
#### Specifying a redirect URI on sign out
```

Also, do we have consistent language on URI vs URL for this feature? E.g. the previous section. Also capitalized vs lower case consistency.

Author: Good catch, I think we can keep it as URI upper case as that's how it's on the Cognito [docs](https://docs.aws.amazon.com/cognito/latest/developerguide/authorization-endpoint.html#:~:text=be%20an%20absolute-,URI,-.).

> #### Option to specify a redirect URL upon signOut

Reviewer 2: I think this section can be moved to the signOut API section on gen2 docs instead. The changes we are making are affecting the redirect experience during signOut.

Author: Not opposed to that but Gen2 has no section on signInWithRedirect so there is no push to use that API on the Gen2 docs yet and this is particularly only useful with HostedUI in picture. Putting it in the general signOut page is unnecessary without context on signInWithRedirect.

Author: Discussed offline, added it to Gen2 docs

> If you have multiple redirect urls configured, you may choose to override the default behavior of selecting a redirect url and provide the one of your choosing when calling `signOut`. The provided redirect url should match at least one of the configured redirect urls. If no redirect url is provided to `signOut`, the first item from the the configured redirect urls list that does not contain a http nor https prefix will be picked.

Reviewer 2: is this text filtered for RN docs only?

Author: Yes

> ```ts

Reviewer 2: is this code snippet filtered for RN docs only?

Author: Yes

import { Amplify } from 'aws-amplify';
import { signOut } from 'aws-amplify/auth';

Amplify.configure({
  Auth: {
    Cognito: {
      loginWith: {
        oauth: {
          redirectSignIn: [
            'myDevApp://'
          ],
          redirectSignOut: [
            'myDevApp://',
            'myProdApp://'
          ],
          ...oauthConfig
        }
      },
      ...userPoolConfig
    }
  }
});

function handleSignOut() {
  signOut({
    global: false,
    oauth: {

>       redirectUrl: 'myDevApp://'

Reviewer 3: Would `myProdApp://` be a better example here as `myDevApp://` would get selected by default anyway?

Author: I think a better example would be to use a different url altogether so updated it to show that.

    }
  });
}

```

> <Callout> Irrespective of whether a `redirectUrl` is provided to the `signOut`, an item that does not contain a http or https string is expected to be present in the configured redirect urls list. This is because iOS requires we provide an appScheme when creating the web session. </Callout>

Reviewer 3: ```suggestion
<Callout> Irrespective of whether a `redirectUrl` is provided to `signOut`, a URL that does not contain http or https is expected to be present in the configured redirect URL list. This is because iOS requires an appScheme when creating the web session. </Callout>
```

> <Callout> Irrespective of whether a `redirectUrl` is provided to the `signOut`, an item that does not contain a http or https string is expected to be present in the configured redirect urls list. This is because iOS requires we provide an appScheme when creating the web session. </Callout>

Reviewer 2: just making sure this text is filtered for RN docs only?

Author: yes

Author: Its within the <InlineFilter filters={["react-native"]}> block
